src/manage_nsfs/nc_lifecycle.js

```diff
@@ -137,13 +137,20 @@ async function run_lifecycle(config_fs, disable_service_validation) {
  */
 async function process_buckets(config_fs, bucket_names, system_json) {
     const buckets_concurrency = 10; // TODO - think about it
-    await P.map_with_concurrency(buckets_concurrency, bucket_names, async bucket_name =>
-        await _call_op_and_update_status({
-            bucket_name,
-            op_name: TIMED_OPS.PROCESS_BUCKET,
-            op_func: async () => process_bucket(config_fs, bucket_name, system_json)
-        })
-    );
+    let is_all_buckets_finished = false;
+    while (!is_all_buckets_finished) {
+        await P.map_with_concurrency(buckets_concurrency, bucket_names, async bucket_name =>
+            await _call_op_and_update_status({
+                bucket_name,
+                op_name: TIMED_OPS.PROCESS_BUCKET,
+                op_func: async () => process_bucket(config_fs, bucket_name, system_json)
+            })
+        );
+        is_all_buckets_finished = Object.values(lifecycle_run_status.buckets_statuses).reduce(
+            (acc, bucket) => acc && (bucket.state?.is_finished),
+            true
+        );
```

Comment:
do you mind setting is_finished on the run as well? 
so we can know if we processed all the buckets or not